pyqtgraph/graphicsItems/InfiniteLine.py

```diff
@@ -174,24 +181,37 @@ def setValue(self, v):
         #else:
             #print "ignore", change
         #return GraphicsObject.itemChange(self, change, val)
-                
+
+    def viewTransformChanged(self):
+        self._boundingRect = None
+        self._line = None
+        GraphicsObject.viewTransformChanged(self)
+
+    def viewChanged(self, view, oldView):
```

Comment:
Overriding `viewChanged` should be unnecessary--this will automatically call `viewTransformChanged` as well.